app/models/user.rb

```diff
@@ -142,56 +146,33 @@ def administered_order_details
     OrderDetail.where(account_id: Account.administered_by(self))
   end
 
-  def full_name
-    if first_name.nil? && last_name.nil?
-      username
-    else
-      full = ""
-      full += first_name unless first_name.nil?
-      full += " " unless first_name.nil? || last_name.nil?
-      full += last_name unless last_name.nil?
-      full
-    end
+  def full_name(suspended_label: true)
+    Users::NamePresenter.new(self, suspended_label: suspended_label).full_name
   end
+  alias to_s full_name
+  alias name full_name
 
-  def last_first_name
-    "#{last_name}, #{first_name}" + deactivated_string
-  end
-
-  def to_s
-    full_name + deactivated_string
-  end
-  alias name to_s
-
-  def deactivated_string
-    if active?
-      ""
-    else
-      " (#{self.class.human_attribute_name(:deactivated)})"
-    end
+  def last_first_name(suspended_label: true)
+    Users::NamePresenter.new(self, suspended_label: suspended_label).last_first_name
   end
 
   # Devise uses this method for determining if a user is allowed to log in. It
-  # also gets called on each request, so if a user gets deactivated, they'll be
+  # also gets called on each request, so if a user gets suspended, they'll be
   # kicked out of their session.
   def active_for_authentication?
     super && active?
   end
 
-  def deactivate
-    update_attribute(:deactivated_at, deactivated_at || Time.current)
-  end
-
-  def activate
-    update_attribute(:deactivated_at, nil)
+  def active?
```

Comment:
Should this and `self.active` below be renamed as `unsuspended` as well? Also, how come we're using a class method below and not a scope?

Reply:
I actually think `active` is accurate here and is definitely less clunky than `unsuspended` (and semantically, I wouldn't consider something unsuspended unless it had been first suspended). It definitely is more of an meaningful term when being used to determine if you are allowed to log in or not.

It could become broader at some point, but right now, it essentially means not suspended.

As far as the class method, that's how we used to do a lot of our scopes. We can definitely make it a scope, though.

Reply:
We mix and match. In User, it seems like we always use the class method. I could go either way.

Reply:
Okay, if we're viewing this as a broader term that may eventually be expanded, I think this is fine then.